apple/macbook-pro/12-1/default.nix

```
@@ -7,6 +7,29 @@
     <nixpkgs/nixos/modules/hardware/network/broadcom-43xx.nix>
   ];
 
+  powerManagement = {
+    # enable gradually increasing/decreasing CPU frequency, rather than using
+    # "powersave", which would keep CPU frequency at 0.8GHz.
+    cpuFreqGovernor = lib.mkDefault "conservative";
+  };
+
+  # brcmfmac wireless module being loaded during hibernation would not let a
+  # successful resume: https://bugzilla.kernel.org/show_bug.cgi?id=101681#c116
+  systemd.services.pre-sleep-brcmfmac = {
+    description = "Pre-Sleep Actions";
+    wantedBy = [ "sleep.target" ];
+    before = [ "sleep.target" ];
```

Reading the manpage: https://www.freedesktop.org/software/systemd/man/systemd.special.html#
it should be those targets. but please test this part:

```suggestion
    wantedBy = [ "hibernate.target" "hybrid-sleep.target" ];
    before = [ "hibernate.target" "hybrid-sleep.target" ];
```